File: solutions/_1980_find_unique_binary_string/solution.py
from typing import List


class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        n = len(nums)
        for i in range(1 << n):
            s = bin(i)[2:].zfill(n)
            if s not in nums:
                return s
        return ""


# https://leetcode.com/problems/find-unique-binary-string/

File: solutions/_1980_find_unique_binary_string/solution.js
/**
 * @param {string[]} nums
 * @return {string}
 */
var findDifferentBinaryString = function (nums) {
  const n = nums.length;
  const set = new Set(nums);
  for (let i = 0; i < 1 << n; i++) {
    const str = i.toString(2).padStart(n, "0");
    if (!set.has(str)) {
      return str;
    }
  }
  return "";
};

module.exports = { findDifferentBinaryString };

/*
https://leetcode.com/problems/find-unique-binary-string/
*/
